pandas/core/index.py

```diff
@@ -3299,6 +3322,415 @@ def _wrap_joined_index(self, joined, other):
 Int64Index._add_logical_methods()
 
 
+class RangeIndex(Int64Index):
+
+    """
+    Immutable Index implementing an monotonic range. RangeIndex is a 
+    memory-saving special case of `Int64Index` limited to representing 
+    monotonic ranges.
+
+    Parameters
+    ----------
+    start : int (default: 0)
+    stop : int (default: 0)
+    step : int (default: 1)
+    name : object, optional
+        Name to be stored in the index
+    """
+
+    _typ = 'rangeindex'
+    _engine_type = _index.Int64Engine
+    _attributes = ['name', 'start', 'stop', 'step']
+
+    def __new__(cls, start=None, stop=None, step=None, name=None, fastpath=False):
+        if fastpath:
+            return cls._simple_new(start, stop, step, name=name)
+
+        # RangeIndex() constructor
+        if start is None and stop is None and step is None:
+            return cls._simple_new(0, 0, 1, name=name)
+
+        # sort the arguments depending on which are provided
+        if step is None:
+            step = 1
+        if stop is None:
+            stop = start
+            start = 0
+
+        # check validity of inputs
+        start = cls._ensure_int(start)
```

Comment:
instead of writing ``_ensure_int``, use ``_ensure_int64`` (from ``core.common``)